File: core/src/main/java/ir/logicfan/core/data/network/error/ErrorHandlingInterceptor.java
package ir.logicfan.core.data.network.error;

import android.content.Context;
import ir.logicfan.core.util.Connectivity;
import okhttp3.Interceptor;
import okhttp3.Response;

import javax.inject.Inject;
import java.io.IOException;

public class ErrorHandlingInterceptor implements Interceptor {
    private Context context;

    @Inject
    public ErrorHandlingInterceptor(Context context) {
        this.context = context;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        if (!Connectivity.isUserOnline(context))
            throw new UserOfflineException();
        return chain.proceed(chain.request());
    }
}

File: core/src/main/java/ir/logicfan/core/data/network/error/ErrorHandlingInterceptor.kt
package ir.logicfan.core.data.network.error

import android.content.Context
import ir.logicfan.core.util.Connectivity
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import javax.inject.Inject

class ErrorHandlingInterceptor @Inject
constructor(private val context: Context) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        if (!Connectivity.isUserOnline(context)) {
            throw UserOfflineException()
        }
        return chain.proceed(chain.request())
    }
}
